src/rapids_pre_commit_hooks/copyright.py

```diff
@@ -0,0 +1,437 @@
+
+    return changed_files
+
+
+def find_blob(tree, filename):
+    d1, d2 = os.path.split(filename)
+    split = [d2]
+    while d1:
+        d1, d2 = os.path.split(d1)
+        split.insert(0, d2)
+
+    while len(split) > 1:
+        component = split.pop(0)
+        try:
+            tree = next(t for t in tree.trees if t.name == component)
+        except StopIteration:
+            return None
+
+    try:
+        return next(blob for blob in tree.blobs if blob.name == split[0])
+    except StopIteration:
+        return None
+
+
+def get_file_last_modified(commit, filename):
+    blob = find_blob(commit.tree, filename)
+    if not blob:
+        return (None, None)
+
+    queue = [(commit, blob)]
+    last_modified = None
+    checked = set()
+
+    while queue:
+        commit, blob = queue.pop(0)
+        if (commit.hexsha, blob.path) in checked:
+            continue
+        checked.add((commit.hexsha, blob.path))
+        all_modified = True
+
+        for parent_commit in commit.parents:
+
+            def compare_files(old_blob):
+                nonlocal all_modified
+
+                if old_blob.hexsha == blob.hexsha:
+                    # Same file contents
+                    all_modified = False
+                    queue.append((parent_commit, old_blob))
+                else:
+                    # Different file contents, but non-copyright-header content might be
+                    # the same
+                    old_content, new_content = (
+                        old_blob.data_stream.read().decode(),
+                        blob.data_stream.read().decode(),
+                    )
+                    old_copyright_matches, new_copyright_matches = match_copyright(
+                        old_content
+                    ), match_copyright(new_content)
+
+                    if strip_copyright(
+                        old_content, old_copyright_matches
+                    ) == strip_copyright(new_content, new_copyright_matches):
+                        all_modified = False
+                        queue.append((parent_commit, old_blob))
+
+            if parent_blob := find_blob(parent_commit.tree, blob.path):
+                compare_files(parent_blob)
+            else:
+                diffs = parent_commit.diff(
+                    other=commit,
+                    find_copies=True,
+                    find_copies_harder=True,
+                    find_renames=True,
+                )
+                diff = next(diff for diff in diffs if diff.b_path == blob.path)
+                if diff.change_type != "A":
+                    compare_files(diff.a_blob)
+
+        if all_modified:
+            if (
+                not last_modified
+                or commit.committed_datetime > last_modified[0].committed_datetime
+            ):
+                last_modified = (commit, blob)
+
+    assert last_modified
+    return last_modified
+
+
+def apply_batch_copyright_check(repo, linter):
```

Comment:
> `find -type f | xargs verify-copyright --batch`

Keep in mind that this script doesn't do any filtering by itself based on file type - it leaves that up to `pre-commit`. This will run it on files that it wasn't meant to be run on.